Add support for OBS stream status and stream key update.

lamaral opened the pull request:
Lange, 

I extended the nodecg-utility-obs to obtain information if OBS is streaming and added the option to change the stream key from NodeCG. 

I hope the code is useful. If it's not up to the standards, feel free to either direct me on how to enhance or fix as you would like.

lamaral commented:
I realized the documentation is missing. I'll take care of it tomorrow.

Lange commented on packages/nodecg-utility-obs/index.js:
```diff
@@ -133,6 +135,39 @@ class OBSUtility extends OBSWebSocket {
+			callback();
+		});
+
+        nodecg.listenFor(`${namespace}:setStreamingKey`, (key, callback = function () {}) => {
```
Change `setStreamingKey` to `setStreamKey`.

Lange commented on packages/nodecg-utility-obs/index.js:
```diff
@@ -133,6 +135,39 @@ class OBSUtility extends OBSWebSocket {
+            try {
+                this.SetStreamSettings({'settings': {'key': key}});
+            } catch (error) {
+                log.error('Error setting the streaming key:', error);
```
Change `streaming` to `stream`.

Lange commented on packages/nodecg-utility-obs/index.js:
```diff
@@ -370,4 +417,4 @@ function buildSchemaPath(schemaName) {
 }
 
-module.exports = OBSUtility;
+module.exports = OBSUtility;
```
The newline at the end of the file should not be removed.

lamaral commented:
859b7b2c5fd18e2e2bce3cd66916426486c06939 contains the requested changes and updated documentation. 

Please double check if I changed all space indentation to tabs.

Lange commented:
Looks like you've still got some formatting issues. You can run `npm test` locally from the root `nodecg-obs` dir to run the tests and make sure everything is ok before you push.

lamaral commented:
I've struggled quite a lot trying to get it right, but the lint is asking for a change that breaks the obs-websocket-js library. 

The lint complained I should use the shorter `settings.key` but if I do, the library complains that settings wasn't defined.

If you wanna close this PR, I don't mind.

Hoishin commented:
what exactly is the eslint error?

lamaral commented:
```
<path>/nodecg-obs/packages/nodecg-utility-obs/index.js
  162:29  error  Unnecessarily quoted property 'settings' found  quote-props
  162:42  error  Unnecessarily quoted property 'key' found       quote-props
  162:42  error  Expected property shorthand                     object-shorthand
```

Hoishin commented:
Run 
```
npm run pretest -- --fix
```

And see if the same error happens after that.
